src/data/nat/sqrt.lean

```diff
@@ -191,4 +191,20 @@ theorem exists_mul_self (x : ℕ) :
   (∃ n, n * n = x) ↔ sqrt x * sqrt x = x :=
 ⟨λ ⟨n, hn⟩, by rw [← hn, sqrt_eq], λ h, ⟨sqrt x, h⟩⟩
 
+theorem sqrt_lt_succ (n : ℕ) : sqrt n * sqrt n < n + 1 :=
+lt_succ_iff.mpr (sqrt_le _)
+
+theorem succ_le_succ_sqrt (n : ℕ) : n + 1 ≤ (sqrt n + 1) * (sqrt n + 1) :=
+le_of_pred_lt (lt_succ_sqrt _)
+
+/-- There are no perfect squares strictly between n² and (m+1)² -/
+theorem no_middle_square {n m : ℕ} (hl : m * m < n) (hr : n < (m + 1) * (m + 1)):
```

Comment:
```suggestion
theorem not_exists_sq {n m : ℕ} (hl : m * m < n) (hr : n < (m + 1) * (m + 1)):
```
or maybe `not_exists_sq_between`

Reply:
Renamed as suggested and fixed typo in doc comment.